<!DOCTYPE html>
<html lang="" xml:lang="">
<head>

  <meta charset="utf-8" />
  <meta http-equiv="X-UA-Compatible" content="IE=edge" />
  <title> 2 Basic single cell pipeline | Report title</title>
  <meta name="description" content=" 2 Basic single cell pipeline | Report title" />
  <meta name="generator" content="bookdown 0.17 and GitBook 2.6.7" />

  <meta property="og:title" content=" 2 Basic single cell pipeline | Report title" />
  <meta property="og:type" content="book" />
  
  
  
  <meta name="github-repo" content="git@github.com:PSL-winter-school-2021/01-monday-workshop-rna.git" />

  <meta name="twitter:card" content="summary" />
  <meta name="twitter:title" content=" 2 Basic single cell pipeline | Report title" />
  
  
  

<meta name="author" content="Lorette Noiret, Nathalie Lehmann, Eric Van Leen" />


<meta name="date" content="2021-02-05" />

  <meta name="viewport" content="width=device-width, initial-scale=1" />
  <meta name="apple-mobile-web-app-capable" content="yes" />
  <meta name="apple-mobile-web-app-status-bar-style" content="black" />
  
  
<link rel="prev" href="working-with-seurat.html"/>
<link rel="next" href="identify-cell-populations.html"/>
<script src="libs/jquery-2.2.3/jquery.min.js"></script>
<link href="libs/gitbook-2.6.7/css/style.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-table.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-bookdown.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-highlight.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-search.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-fontsettings.css" rel="stylesheet" />
<link href="libs/gitbook-2.6.7/css/plugin-clipboard.css" rel="stylesheet" />











<style type="text/css">
code.sourceCode > span { display: inline-block; line-height: 1.25; }
code.sourceCode > span { color: inherit; text-decoration: inherit; }
code.sourceCode > span:empty { height: 1.2em; }
.sourceCode { overflow: visible; }
code.sourceCode { white-space: pre; position: relative; }
pre.sourceCode { margin: 0; }
@media screen {
div.sourceCode { overflow: auto; }
}
@media print {
code.sourceCode { white-space: pre-wrap; }
code.sourceCode > span { text-indent: -5em; padding-left: 5em; }
}
pre.numberSource code
  { counter-reset: source-line 0; }
pre.numberSource code > span
  { position: relative; left: -4em; counter-increment: source-line; }
pre.numberSource code > span > a:first-child::before
  { content: counter(source-line);
    position: relative; left: -1em; text-align: right; vertical-align: baseline;
    border: none; display: inline-block;
    -webkit-touch-callout: none; -webkit-user-select: none;
    -khtml-user-select: none; -moz-user-select: none;
    -ms-user-select: none; user-select: none;
    padding: 0 4px; width: 4em;
    color: #aaaaaa;
  }
pre.numberSource { margin-left: 3em; border-left: 1px solid #aaaaaa;  padding-left: 4px; }
div.sourceCode
  {   }
@media screen {
code.sourceCode > span > a:first-child::before { text-decoration: underline; }
}
code span.al { color: #ff0000; font-weight: bold; } /* Alert */
code span.an { color: #60a0b0; font-weight: bold; font-style: italic; } /* Annotation */
code span.at { color: #7d9029; } /* Attribute */
code span.bn { color: #40a070; } /* BaseN */
code span.bu { } /* BuiltIn */
code span.cf { color: #007020; font-weight: bold; } /* ControlFlow */
code span.ch { color: #4070a0; } /* Char */
code span.cn { color: #880000; } /* Constant */
code span.co { color: #60a0b0; font-style: italic; } /* Comment */
code span.cv { color: #60a0b0; font-weight: bold; font-style: italic; } /* CommentVar */
code span.do { color: #ba2121; font-style: italic; } /* Documentation */
code span.dt { color: #902000; } /* DataType */
code span.dv { color: #40a070; } /* DecVal */
code span.er { color: #ff0000; font-weight: bold; } /* Error */
code span.ex { } /* Extension */
code span.fl { color: #40a070; } /* Float */
code span.fu { color: #06287e; } /* Function */
code span.im { } /* Import */
code span.in { color: #60a0b0; font-weight: bold; font-style: italic; } /* Information */
code span.kw { color: #007020; font-weight: bold; } /* Keyword */
code span.op { color: #666666; } /* Operator */
code span.ot { color: #007020; } /* Other */
code span.pp { color: #bc7a00; } /* Preprocessor */
code span.sc { color: #4070a0; } /* SpecialChar */
code span.ss { color: #bb6688; } /* SpecialString */
code span.st { color: #4070a0; } /* String */
code span.va { color: #19177c; } /* Variable */
code span.vs { color: #4070a0; } /* VerbatimString */
code span.wa { color: #60a0b0; font-weight: bold; font-style: italic; } /* Warning */
</style>

</head>

<body>



  <div class="book without-animation with-summary font-size-2 font-family-1" data-basepath=".">

    <div class="book-summary">
      <nav role="navigation">

<ul class="summary">
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html"><i class="fa fa-check"></i>Introdution</a><ul>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#sources-of-inspiration"><i class="fa fa-check"></i>Sources of inspiration</a></li>
<li class="chapter" data-level="" data-path="introdution.html"><a href="introdution.html#bibliography"><i class="fa fa-check"></i>Bibliography</a></li>
</ul></li>
<li class="chapter" data-level="1" data-path="working-with-seurat.html"><a href="working-with-seurat.html"><i class="fa fa-check"></i><b>1</b> Working with Seurat</a><ul>
<li class="chapter" data-level="1.1" data-path="working-with-seurat.html"><a href="working-with-seurat.html#load-the-library"><i class="fa fa-check"></i><b>1.1</b> Load the library</a></li>
<li class="chapter" data-level="1.2" data-path="working-with-seurat.html"><a href="working-with-seurat.html#import-the-data"><i class="fa fa-check"></i><b>1.2</b> Import the data</a></li>
<li class="chapter" data-level="1.3" data-path="working-with-seurat.html"><a href="working-with-seurat.html#explore-the-seurat-data-structure"><i class="fa fa-check"></i><b>1.3</b> Explore the Seurat data structure</a></li>
</ul></li>
<li class="chapter" data-level="2" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html"><i class="fa fa-check"></i><b>2</b> Basic single cell pipeline</a><ul>
<li class="chapter" data-level="2.1" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#remove-low-quality-cells"><i class="fa fa-check"></i><b>2.1</b> Remove “low quality cells”</a><ul>
<li class="chapter" data-level="2.1.1" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#number-of-genes-detected-per-cell"><i class="fa fa-check"></i><b>2.1.1</b> Number of genes detected per cell</a></li>
<li class="chapter" data-level="2.1.2" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#number-of-umis-per-cell"><i class="fa fa-check"></i><b>2.1.2</b> Number of UMIs per cell</a></li>
<li class="chapter" data-level="2.1.3" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#percentage-of-mitochondrial-genes"><i class="fa fa-check"></i><b>2.1.3</b> Percentage of mitochondrial genes</a></li>
<li class="chapter" data-level="2.1.4" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#group-exercice-on-filtering"><i class="fa fa-check"></i><b>2.1.4</b> Group exercice on filtering</a></li>
</ul></li>
<li class="chapter" data-level="2.2" data-path="basic-single-cell-pipeline.html"><a href="basic-single-cell-pipeline.html#normalize-the-expression-in-cells"><i class="fa fa-check"></i><b>2.2</b> Normalize the expression in cells</a></li>
</ul></li>
<li class="chapter" data-level="3" data-path="identify-cell-populations.html"><a href="identify-cell-populations.html"><i class="fa fa-check"></i><b>3</b> Identify cell populations</a><ul>
<li class="chapter" data-level="3.1" data-path="identify-cell-populations.html"><a href="identify-cell-populations.html#select-highly-variables-hvg"><i class="fa fa-check"></i><b>3.1</b> Select highly variables (hvg)</a></li>
<li class="chapter" data-level="3.2" data-path="identify-cell-populations.html"><a href="identify-cell-populations.html#dimension-reduction"><i class="fa fa-check"></i><b>3.2</b> Dimension Reduction</a><ul>
<li class="chapter" data-level="3.2.1" data-path="identify-cell-populations.html"><a href="identify-cell-populations.html#scaling"><i class="fa fa-check"></i><b>3.2.1</b> Scaling</a></li>
</ul></li>
<li class="chapter" data-level="3.3" data-path="identify-cell-populations.html"><a href="identify-cell-populations.html#cluster-cells"><i class="fa fa-check"></i><b>3.3</b> Cluster cells</a></li>
</ul></li>
<li class="chapter" data-level="" data-path="references.html"><a href="references.html"><i class="fa fa-check"></i>References</a></li>
<li class="chapter" data-level="4" data-path="session-info.html"><a href="session-info.html"><i class="fa fa-check"></i><b>4</b> Session info</a></li>
</ul>

      </nav>
    </div>

    <div class="book-body">
      <div class="body-inner">
        <div class="book-header" role="navigation">
          <h1>
            <i class="fa fa-circle-o-notch fa-spin"></i><a href="./">Report title</a>
          </h1>
        </div>

        <div class="page-wrapper" tabindex="-1" role="main">
          <div class="page-inner">

            <section class="normal" id="section-">
<div id="basic-single-cell-pipeline" class="section level1">
<h1><span class="header-section-number"> 2</span> Basic single cell pipeline</h1>
<p>The steps below encompass the standard pre-processing workflow for scRNA-seq data.</p>
<ul>
<li>Filter out low quality cells : undersequenced cells, debris (broken cells or floating pieces of RNA) and multiplets (several cells in one droplets)</li>
<li>Normalize the expression to reduce the sequencing biais.</li>
</ul>
<div id="remove-low-quality-cells" class="section level2">
<h2><span class="header-section-number">2.1</span> Remove “low quality cells”</h2>
<p>The are a few metrics commonly used to filter out low quality cells :</p>
<ul>
<li>the number of unique genes detected in each cell :
<ul>
<li>cells with low gene count : helps to remove empty droplets, debris and undersequenced cells.</li>
<li>cells with high gene count : these cells are more likely to be cell doublets or multiplets. The number of multiplets expected in your sample is directly correlated with the number of cells you loaded in 10X. The more cells you loaded, the higher the chance is to have 2 or more cells in one droplet.</li>
</ul></li>
<li>total number of UMI detected within a cell</li>
<li>the percentage of reads that map to the mitochondrial genome : low-quality / dying cells often exhibit extensive mitochondrial contamination</li>
</ul>
<p>Seurat allows you to easily explore QC (quality control) metrics and filter cells based on any user-defined criteria using the slot ‘meta.data’.</p>
<div id="number-of-genes-detected-per-cell" class="section level3">
<h3><span class="header-section-number">2.1.1</span> Number of genes detected per cell</h3>
<div class="sourceCode" id="cb17"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb17-1"><a href="basic-single-cell-pipeline.html#cb17-1"></a><span class="kw">barplot</span>(<span class="kw">sort</span>(mydata<span class="op">$</span>nFeature_RNA), <span class="dt">main=</span><span class="st">&quot;Number of genes detected/Cell&quot;</span>, <span class="dt">names.arg =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
<div class="sourceCode" id="cb18"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb18-1"><a href="basic-single-cell-pipeline.html#cb18-1"></a><span class="kw">quantile</span>(mydata<span class="op">$</span>nFeature_RNA)</span></code></pre></div>
<pre><code>##     0%    25%    50%    75%   100% 
##  235.0  705.5 1312.0 2593.5 6322.0</code></pre>
</div>
<div id="number-of-umis-per-cell" class="section level3">
<h3><span class="header-section-number">2.1.2</span> Number of UMIs per cell</h3>
<div class="sourceCode" id="cb20"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb20-1"><a href="basic-single-cell-pipeline.html#cb20-1"></a><span class="kw">barplot</span>(<span class="kw">sort</span>(mydata<span class="op">$</span>nCount_RNA), <span class="dt">main=</span><span class="st">&quot;Number of UMI/Cell&quot;</span>,<span class="dt">names.arg =</span> <span class="st">&quot;&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-9-1.png" width="672" /></p>
<div class="sourceCode" id="cb21"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb21-1"><a href="basic-single-cell-pipeline.html#cb21-1"></a><span class="kw">VlnPlot</span>(mydata, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>), <span class="dt">ncol =</span> <span class="dv">2</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-9-2.png" width="672" /></p>
<div class="sourceCode" id="cb22"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb22-1"><a href="basic-single-cell-pipeline.html#cb22-1"></a><span class="kw">FeatureScatter</span>(mydata, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-9-3.png" width="672" /></p>
<p>You aim while looking at these graphs is to define thresholds/cutoff that you will aply to filter cell out. Try to identify the cells that behave differently from the main population.</p>
</div>
<div id="percentage-of-mitochondrial-genes" class="section level3">
<h3><span class="header-section-number">2.1.3</span> Percentage of mitochondrial genes</h3>
<p>To calculate the perentage of mitochonrial genes expressed in a cell, you must first provide a list of mitochondrial genes. In the paper, the authors did not provide their list. As a quick replacement, we will consider that mitochondrial genes are all the genes that starts with the pattern ‘mt:’. For a more refine analysis, you can use flybase or another source and carefully identify mitochondrial genes.</p>
<p>We then use the PercentageFeatureSet function, which calculates the percentage of counts originating from a set of features.</p>
<div class="sourceCode" id="cb23"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb23-1"><a href="basic-single-cell-pipeline.html#cb23-1"></a>mydata[[<span class="st">&quot;percent.mt&quot;</span>]] &lt;-<span class="st"> </span><span class="kw">PercentageFeatureSet</span>(mydata, <span class="dt">pattern =</span> <span class="st">&quot;^mt:&quot;</span>)</span>
<span id="cb23-2"><a href="basic-single-cell-pipeline.html#cb23-2"></a></span>
<span id="cb23-3"><a href="basic-single-cell-pipeline.html#cb23-3"></a><span class="co"># Alternative : if you have a list of mitochondrial genes</span></span>
<span id="cb23-4"><a href="basic-single-cell-pipeline.html#cb23-4"></a><span class="co">#list = c(&quot;mTerf3&quot;,&quot;Mtpalpha&quot; ,&quot;mTTF&quot;)</span></span>
<span id="cb23-5"><a href="basic-single-cell-pipeline.html#cb23-5"></a><span class="co">#mydata[[&quot;percent.mt2&quot;]] &lt;- PercentageFeatureSet(mydata, features=list_mito )</span></span></code></pre></div>
<div class="sourceCode" id="cb24"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb24-1"><a href="basic-single-cell-pipeline.html#cb24-1"></a><span class="kw">VlnPlot</span>(mydata, <span class="dt">features =</span> <span class="st">&quot;percent.mt&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-11-1.png" width="672" /></p>
<div class="sourceCode" id="cb25"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb25-1"><a href="basic-single-cell-pipeline.html#cb25-1"></a><span class="kw">FeatureScatter</span>(mydata, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-11-2.png" width="672" /></p>
<div class="sourceCode" id="cb26"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb26-1"><a href="basic-single-cell-pipeline.html#cb26-1"></a><span class="kw">FeatureScatter</span>(mydata, <span class="dt">feature1 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-11-3.png" width="672" /></p>
</div>
<div id="group-exercice-on-filtering" class="section level3">
<h3><span class="header-section-number">2.1.4</span> Group exercice on filtering</h3>
<p>On of the most difficult parts of single cells pre-processing is to decide the threshold to define low and high quality cells.
From the various graphs, decide the levels for low high quality cells dependant on the number of genes expressed per cell, the total UMIs per cell and the percentage of mitochonrial genes.</p>
<p>Discuss as a group.</p>
<div class="sourceCode" id="cb27"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb27-1"><a href="basic-single-cell-pipeline.html#cb27-1"></a><span class="co"># DEFINE THE CUTOFF VALUES AND VISUALIZE THE RESULTS</span></span>
<span id="cb27-2"><a href="basic-single-cell-pipeline.html#cb27-2"></a>minGene    =<span class="st"> </span><span class="dv">500</span></span>
<span id="cb27-3"><a href="basic-single-cell-pipeline.html#cb27-3"></a>maxGene    =<span class="st"> </span><span class="dv">5800</span></span>
<span id="cb27-4"><a href="basic-single-cell-pipeline.html#cb27-4"></a>minUMI     =<span class="st"> </span><span class="dv">12500</span></span>
<span id="cb27-5"><a href="basic-single-cell-pipeline.html#cb27-5"></a>maxUMI     =<span class="st"> </span><span class="dv">60000</span></span>
<span id="cb27-6"><a href="basic-single-cell-pipeline.html#cb27-6"></a>maxpct_mt  =<span class="st"> </span><span class="dv">18</span>  </span>
<span id="cb27-7"><a href="basic-single-cell-pipeline.html#cb27-7"></a></span>
<span id="cb27-8"><a href="basic-single-cell-pipeline.html#cb27-8"></a><span class="co"># the function &quot;subset&quot; helps you to filter the cells</span></span>
<span id="cb27-9"><a href="basic-single-cell-pipeline.html#cb27-9"></a><span class="co"># we create a new seurat object containing the filtred cells</span></span>
<span id="cb27-10"><a href="basic-single-cell-pipeline.html#cb27-10"></a>mydata_filtrd &lt;-<span class="st"> </span><span class="kw">subset</span>(mydata, <span class="dt">subset =</span> nFeature_RNA <span class="op">&gt;</span><span class="st"> </span>minGene <span class="op">&amp;</span><span class="st"> </span>nFeature_RNA <span class="op">&lt;</span><span class="st"> </span>maxGene <span class="op">&amp;</span><span class="st"> </span>nCount_RNA <span class="op">&gt;</span><span class="st"> </span>minUMI  <span class="op">&amp;</span><span class="st"> </span>nCount_RNA <span class="op">&lt;</span><span class="st"> </span>maxUMI <span class="op">&amp;</span><span class="st"> </span>percent.mt <span class="op">&lt;</span><span class="st"> </span>maxpct_mt)</span>
<span id="cb27-11"><a href="basic-single-cell-pipeline.html#cb27-11"></a></span>
<span id="cb27-12"><a href="basic-single-cell-pipeline.html#cb27-12"></a><span class="kw">VlnPlot</span>(mydata_filtrd, <span class="dt">features =</span> <span class="kw">c</span>(<span class="st">&quot;nFeature_RNA&quot;</span>, <span class="st">&quot;nCount_RNA&quot;</span>, <span class="st">&quot;percent.mt&quot;</span>), <span class="dt">ncol =</span> <span class="dv">3</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-12-1.png" width="672" /></p>
<div class="sourceCode" id="cb28"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb28-1"><a href="basic-single-cell-pipeline.html#cb28-1"></a><span class="kw">FeatureScatter</span>(mydata_filtrd, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;percent.mt&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-12-2.png" width="672" /></p>
<div class="sourceCode" id="cb29"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb29-1"><a href="basic-single-cell-pipeline.html#cb29-1"></a><span class="kw">FeatureScatter</span>(mydata_filtrd, <span class="dt">feature1 =</span> <span class="st">&quot;nCount_RNA&quot;</span>, <span class="dt">feature2 =</span> <span class="st">&quot;nFeature_RNA&quot;</span>)</span></code></pre></div>
<p><img src="01-monday-workshop-rna_files/figure-html/unnamed-chunk-12-3.png" width="672" /></p>
</div>
</div>
<div id="normalize-the-expression-in-cells" class="section level2">
<h2><span class="header-section-number">2.2</span> Normalize the expression in cells</h2>
<p>After removing unwanted cells from the dataset, the next step is to normalize the data.
The objective is to correct for the sequencing bias.</p>
<p>As we noticed in the previous graphes, the cells do not have the same number of total UMI. This may be due biological differences (some cells express less RNA than other) but is likely to be the result of cell-specific sequencint biais (some cells have been less sequenced than other).</p>
<p>We are going to apply a global scaling method, which supposes that all cells are about the same RNA content. It aims to remove cell-specific systematic biases by scaling expression measures within each cell by a constant factor.
We use a “LogNormalize” method that normalizes the feature expression measurements for each cell by the total expression, multiplies this by a scale factor (here the median total UMI per cell), and log-transforms the result.</p>
<div class="sourceCode" id="cb30"><pre class="sourceCode r"><code class="sourceCode r"><span id="cb30-1"><a href="basic-single-cell-pipeline.html#cb30-1"></a>mydata_filtrd &lt;-<span class="st"> </span><span class="kw">NormalizeData</span>(mydata_filtrd, <span class="dt">normalization.method =</span> <span class="st">&quot;LogNormalize&quot;</span>, <span class="dt">scale.factor =</span> <span class="kw">median</span>(mydata_filtrd<span class="op">$</span>nCount_RNA))</span></code></pre></div>
<p>The resulting data are stored in the ‘data’ slot</p>
<p>Once you have pre-processed your data, you are ready for further analyses : identify sub-population of cells, pseudo-time analyses…</p>
</div>
</div>
            </section>

          </div>
        </div>
      </div>
<a href="working-with-seurat.html" class="navigation navigation-prev " aria-label="Previous page"><i class="fa fa-angle-left"></i></a>
<a href="identify-cell-populations.html" class="navigation navigation-next " aria-label="Next page"><i class="fa fa-angle-right"></i></a>
    </div>
  </div>
<script src="libs/gitbook-2.6.7/js/app.min.js"></script>
<script src="libs/gitbook-2.6.7/js/lunr.js"></script>
<script src="libs/gitbook-2.6.7/js/clipboard.min.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-search.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-sharing.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-fontsettings.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-bookdown.js"></script>
<script src="libs/gitbook-2.6.7/js/jquery.highlight.js"></script>
<script src="libs/gitbook-2.6.7/js/plugin-clipboard.js"></script>
<script>
gitbook.require(["gitbook"], function(gitbook) {
gitbook.start({
"sharing": {
"github": false,
"facebook": true,
"twitter": true,
"linkedin": false,
"weibo": false,
"instapaper": false,
"vk": false,
"all": ["facebook", "twitter", "linkedin", "weibo", "instapaper"]
},
"fontsettings": {
"theme": "white",
"family": "sans",
"size": 2
},
"edit": {
"link": null,
"text": null
},
"history": {
"link": null,
"text": null
},
"view": {
"link": null,
"text": null
},
"download": null,
"toc": {
"collapse": "subsection"
}
});
});
</script>

</body>

</html>
